internal/commands/scan.go

```diff
@@ -597,30 +596,76 @@ func setupScanTypeProjectAndConfig(
 	info["project"].(map[string]interface{})["id"] = projectID
 	// Handle the scan configuration
 	var configArr []interface{}
-	if _, ok := info["config"]; !ok {
-		err = json.Unmarshal([]byte("[]"), &configArr)
-		if err != nil {
-			return err
+	resubmit, _ := cmd.Flags().GetBool("resubmit")
+	if resubmit {
```

Comment:
We can simplify the if/else by extracting the common part for each engine in the end of each block to outside the if/else

Reply:
Done, thanks !